File: solutions/121. Best Time to Buy and Sell Stock.cpp
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        if (prices.size() == 0 || prices.size() == 1)
        {
            return 0;
        }
​
        int min = prices[0];
        int maxprofit;
        for (size_t i = 0; i < prices.size(); i++)
        {
            if (prices[i] < min)
            {
                min = prices[i];
            }
            else if (prices[i] - min > maxprofit)
            {
                maxprofit = prices[i] - min;
            }
        }
        return maxprofit;
    }
};

File: solutions/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minimum = prices[0]
        maximum_profit = 0
​
        for i in prices:
            if i < minimum:
                minimum = i
            elif i - minimum > maximum_profit:
                maximum_profit = i - minimum
        
        return maximum_profit
